src/Toolkit/Data/List/Subset.idr

```diff
@@ -16,9 +17,7 @@ data Subset : (eq   : a -> b -> Type)
   where
     Empty : Subset eq Nil Nil
 
-    EmptyThis : Subset eq Nil xs
-
-    Keep : {eq : a -> b -> Type}
```

Comment:
This is the biggest departure. The presence of `EmptyThis` made implementing `(++)` impossible
without insisting that the middle list should be available at runtime. Given that it can be derived from
`Skip* Empty`, I've dropped it.

It also makes proofs simpler because we don't need to consider this additional constructor anymore.